src/main/java/com/shanebeestudios/skbee/api/recipe/RecipeUtil.java

```diff
@@ -148,6 +157,146 @@ public static void logShapedRecipe(ShapedRecipe recipe) {
         });
     }
 
+    /**
+     * Get a recipe choice based off given argument
+     *
+     * @param object object to get RecipeChoice from
+     * @return RecipeChoice or null if invalid
+     */
+    @Nullable
+    public static RecipeChoice getRecipeChoice(Object object) {
+        if (object == null) return null;
+        if (object instanceof RecipeChoice recipeChoice) return recipeChoice;
+        if (object instanceof ItemStack itemStack) {
+            boolean isSimilar = itemStack.isSimilar(new ItemStack(itemStack.getType()));
+            if (itemStack.getType().isAir()) return null;
+            if (isSimilar) return new MaterialChoice(itemStack.getType());
+            return new ExactChoice(itemStack);
+        } else if (object instanceof ItemType itemType) {
+            ItemStack item = itemType.getRandom();
+            boolean isSimilar = item.isSimilar(new ItemStack(item.getType()));
+            if (item.getType().isAir()) return null;
+            if (isSimilar) {
+                return getMaterialChoice(itemType);
+            } else {
+                return getExactChoice(itemType);
+            }
+        } else if (object instanceof Tag<?> tag) {
+            // Honestly this shouldn't ever be reached unless we add syntax like %recipechoice/itemtype/minecrafttag%
+            return getMaterialChoice(tag);
+        }
+
+        return null;
+    }
+
+    /**
+     * Get an ExactChoice from a list of objects
+     *
+     * @param objects list of objects to get an exact choice from
+     * @return ExactChoice or null if invalid
+     */
+    @Nullable
+    public static ExactChoice getExactChoice(Object... objects) {
+        if (objects == null) return null;
+        List<ItemStack> itemStacks = new ArrayList<>();
+        for (Object object : objects) {
+            if (object instanceof ExactChoice exactChoice) {
+                for (ItemStack itemStack : exactChoice.getChoices()) {
+                    if (itemStacks.contains(itemStack)) continue;
+                    itemStacks.add(itemStack);
+                }
+            } else if (object instanceof ItemStack itemStack) {
+                if (itemStacks.contains(itemStack)) continue;
+                boolean isAir = itemStack.getType().isAir();
+                if (isAir) continue;
```

Comment:
could this just be one line?
```suggestion
                if (itemStack.getType().isAir()) continue;
```
same with the one bellow

Reply:
I think this habit came from when I first redid these where the old code also split them all off into their own fields